Update lucide-icons extension

Sn0wye opened the pull request:
## Description

- Removed stroke red color to mantain the icons as unchanged as possible
- Updated to latest Lucide Icons version
- Updated dependencies
- Added new keyboard shortcuts to `Copy Name to Clipboard`, `Copy SVG to Clipboard` and `Copy Component to Clipboard`

## Screencast
<img width="862" alt="Screenshot 2024-01-13 at 11 16 14 PM" src="https://github.com/raycast/extensions/assets/95551770/879b3ade-ce34-40bf-83be-f562d61743c0">

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder
- [x] Cache response for extension still work offline

gBasil commented:
Given that there have been many pull requests for this extension in regards to updating the icon set, I feel like it would be better to pull the list of icons remotely from somewhere.

Sn0wye commented:
Maybe do a fetch call and some parsing directly from lucide's github repo?

Sn0wye commented:
> Given that there have been many pull requests for this extension in regards to updating the icon set, I feel like it would be better to pull the list of icons remotely from somewhere.

Maybe do a fetch call and some parsing directly from lucide's github repo?

gBasil commented:
> > Given that there have been many pull requests for this extension in regards to updating the icon set, I feel like it would be better to pull the list of icons remotely from somewhere.
> 
> Maybe do a fetch call/octokit and some parsing directly from lucide's github repo?

Something along those lines, yeah. You might want to ask the Lucide folks (GitHub Discussions or Discord) if they have an existing public list of icons in JSON format.

Sn0wye commented:
Implemented a new logic to fetch the icons directly from lucide's website, take a look!

gBasil commented on extensions/lucide-icons/src/index.tsx:
```diff
@@ -1,45 +1,101 @@
+  const [icons, setIcons] = useCachedState<LucideIcon[] | null>("icons", null);
+
+  console.log(data);
+  // const files = useMemo(
+  //   () =>
+  //     readdirSync(join(environment.assetsPath, "icons"))
+  //       .filter((file) => file.endsWith(".svg"))
+  //       .map((file) => {
+  //         return [join(environment.assetsPath, "icons", file), file.split(".")[0]];
+  //       }),
+  //   []
+  // );
+
```
You probably want to remove this.

gBasil commented on extensions/lucide-icons/src/index.tsx:
```diff
@@ -1,45 +1,101 @@
 
+  const [icons, setIcons] = useCachedState<LucideIcon[] | null>("icons", null);
+
+  console.log(data);
```
And this `console.log` should also probably be removed.

gBasil commented on extensions/lucide-icons/src/index.tsx:
```diff
@@ -1,45 +1,101 @@
+        .then((res) => res.json() as Promise<Record<string, unknown>>)
+        .then((data) => Object.keys(data));
+
+      const promises = response.map(async (iconName) => {
```
How long does it take to fetch all of the icons? The SVG data is already present, albeit in a parsed form, so it might be better to assemble the data into SVGs without refetching them again.

Sn0wye replied:
is there a way to rerun this bit of code that updates them every once in a while?

gBasil replied:
I don't know of any native ways, but I haven't touched the Raycast developer ecosystem in a while.

pernielsentikaer replied:
You can save a datetime object and compare - could that work?

Sn0wye replied:
thought about saving it on the raycast cache, makes sense.

gBasil commented:
Also, as one last thing, you might want to implement the search changes in https://github.com/raycast/extensions/pull/8324.

pernielsentikaer commented on extensions/lucide-icons/src/index.tsx:
```diff
@@ -1,45 +1,101 @@
   return (
-    <Grid columns={columns} inset={Grid.Inset.Large}>
-      {files.map(([path, name]) => (
+    <Grid columns={columns} inset={Grid.Inset.Large} isLoading={isLoading || !data}>
```
```suggestion
    <Grid columns={columns} inset={Grid.Inset.Large} isLoading={isLoading}>
```

pernielsentikaer replied:
And even maybe an EmptyView that shows something while loading is true?

Sn0wye replied:
Will keep an eye on that

Sn0wye commented:
To improve the icon search as mentioned in #8324 I'd need the metadata for the icons (the .json one), which isn't being obtained neither by https://lucide.dev/api/icon-nodes nor https://lucide.dev/api/icons/${iconName}, so any thoughts on how to get it?

gBasil commented:
@Sn0wye Try https://lucide.dev/api/tags.

Sn0wye commented:
> @Sn0wye Try https://lucide.dev/api/tags.

Nice, will include this in a bit

Sn0wye commented:
- Implemented a new way to fetch the icons, building them locally, which is FAR FASTER than fetching individually and making a lot of requests.
- Implemented keyword based search as mentioned in #8324

pernielsentikaer commented:
It works pretty good here, could you check it out @gBasil

gBasil commented:
> It works pretty good here, could you check it out @gBasil

No. I've been thinking about it, and I don't really want to contribute more of my time and labour than I already have to the closed-source product that is Raycast.

Sn0wye commented:
> > It works pretty good here, could you check it out @gBasil
> 
> No. I've been thinking about it, and I don't really want to contribute more of my time and labour than I already have to the closed-source product that is Raycast.

So, for maintaining this extension from now own, can I take ownership of it and republish?

gBasil commented:
Sure, I guess. Might be better to move custodianship to the Lucide team though.

pernielsentikaer commented:
I made Sn0wye as the author instead, thanks for everything you contributed @gBasil - I hope you stay around 🙂

Feel free to DM me in Slack or elsewhere if you want to discuss

Sn0wye commented:
> Sure, I guess. Might be better to move custodianship to the Lucide team though.

Nice, thank you for your contributions, and feel free to dm me if you need something 😉

gBasil commented:
> I made Sn0wye as the author instead, thanks for everything you contributed @gBasil - I hope you stay around 🙂
> 
> Feel free to DM me in Slack or elsewhere if you want to discuss

I don't use Slack and I'm not familiar with it.

My main reason for not wanting to contribute to Raycast is that Raycast itself isn't open source, but is soliciting contribution through the extension system, which rubs me the wrong way.

pernielsentikaer commented:
@gBasil you can find me on X too, if you want - i'd be happy to take a talk with you on this subject
